mule-extensions-api/src/main/java/org/mule/runtime/extension/api/manifest/ExtensionManifestBuilder.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.manifest;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * A builder to create instances of {@link ExtensionManifest}.
+ * <p>
+ * A new instance of this class should be used per each manifest to be
+ * created. The created instances will be immutable.
```

Comment:
if the result is immutable it means that the builder has to be able to hold temporary state. I think it's better to put this limitation (which is how most people would use it anyway) rather than add complexity by resetting the builder's state